### PR TITLE
[exiv2] update to 0.27.5

### LINKED ISSUE #21754: [exiv2] update to 27.5
Library name:
Exiv2
New version number:
27.5
Other information that may be useful (release notes, etc...)

Release Information and download can be found here:
https://exiv2.org/download.html

The latest version of Exiv2 contains fixes for the following Security Vulnerabilities:
https://nvd.nist.gov/vuln/detail/CVE-2021-37615
https://nvd.nist.gov/vuln/detail/CVE-2021-34335
https://nvd.nist.gov/vuln/detail/CVE-2021-37622
https://nvd.nist.gov/vuln/detail/CVE-2021-37621
https://nvd.nist.gov/vuln/detail/CVE-2021-37620
https://nvd.nist.gov/vuln/detail/CVE-2021-37619
https://nvd.nist.gov/vuln/detail/CVE-2021-37618
https://nvd.nist.gov/vuln/detail/CVE-2021-37616
https://nvd.nist.gov/vuln/detail/CVE-2021-37623
https://nvd.nist.gov/vuln/detail/CVE-2021-34334
https://nvd.nist.gov/vuln/detail/CVE-2021-32815

### LINKED ISSUE #21943: [exiv2] Build error
**Host Environment**
- OS: [Windows 7 x64.]
- Compiler: vs2017

**To Reproduce**
`.\vcpkg.exe install exiv2[nls,png,unicode,video,xmp] --triplet x64-windows`

**Failure logs** 
The following packages will be built and installed:
    exiv2[core,nls,png,unicode,video,xmp]:x64-windows -> 0.27.4#2
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\UPUP\AppData\Local\vcpkg\archives in 30.56 ms. Use --debug to see more details.
Starting package 1/1: exiv2:x64-windows
Building package exiv2[core,nls,png,unicode,video,xmp]:x64-windows...
-- Using cached Exiv2-exiv2-15098f4ef50cc721ad0018218acab2ff06e60beb.tar.gz.
-- Cleaning sources at E:/vcpkg/buildtrees/exiv2/src/ff06e60beb-4354818a55.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/Exiv2-exiv2-15098f4ef50cc721ad0018218acab2ff06e60beb.tar.gz
-- Applying patch fix-ninja-error.patch
-- Using source at E:/vcpkg/buildtrees/exiv2/src/ff06e60beb-4354818a55.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: E:/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe --build . --config Debug --target install --
 -v -j3
    Working Directory: E:/vcpkg/buildtrees/exiv2/x64-windows-dbg
    See logs for more information:
      E:\vcpkg\buildtrees\exiv2\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:106 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/exiv2/portfile.cmake:52 (vcpkg_install_cmake)
  scripts/ports.cmake:142 (include)

Error: Building package exiv2:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+exiv2
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[exiv2]+Build+error
including:
  package: exiv2[core,nls,png,unicode,video,xmp]:x64-windows -> 0.27.4#2
    vcpkg-tool version: 2021-11-24-48b94a6946b8a70abd21529218927fd478d02b6c
    vcpkg-scripts version: 14543ba18 2021-12-08 (11 hours ago)

Additionally, attach any relevant sections from the log files above.

**  **
remove nls is ok.
steps: .\vcpkg.exe install exiv2[png,unicode,video,xmp] --triplet x64-windows

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #21754 
Fix #21943 
Update [exiv2] to the latest version0.27.5
All features have been tested successfullly in the following triplet:

- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/exiv2/fix-ninja-error.patch [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-@@ -215,7 +215,7 @@ else()
+ if (EXIV2_ENABLE_XMP OR EXIV2_ENABLE_EXTERNAL_XMP)
+@@ -224,7 +224,7 @@ else()
  endif()
  
  if( EXIV2_ENABLE_PNG )
--    target_link_libraries( exiv2lib PRIVATE $<BUILD_INTERFACE:${ZLIB_LIBRARIES}>)
-+    target_link_libraries( exiv2lib PRIVATE ${ZLIB_LIBRARIES})
+-	target_link_libraries( exiv2lib PRIVATE ZLIB::ZLIB)
++	target_link_libraries( exiv2lib PRIVATE ${ZLIB_LIBRARIES})
```
Comment (BillyONeal) [inline ports/exiv2/fix-ninja-error.patch:19]: This seems wrong now, why isn't using the modern target solution sufficient?

Comment (BillyONeal) [reply ports/exiv2/fix-ninja-error.patch:19]: I think we should delete the entire patch.

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/exiv2/fix-ninja-error.patch (hunk 1 above)
Comment (JackBoosY) [reply ports/exiv2/fix-ninja-error.patch:19]: Agreed.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/exiv2/fix-find_expat.patch [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+index ec3a43f..b4cc2b5 100644
+--- a/cmake/findDependencies.cmake
++++ b/cmake/findDependencies.cmake
+@@ -42,7 +42,8 @@ if (EXIV2_ENABLE_XMP AND EXIV2_ENABLE_EXTERNAL_XMP)
+     message(FATAL_ERROR "EXIV2_ENABLE_XMP AND EXIV2_ENABLE_EXTERNAL_XMP are mutually exclusive.  You can only choose one of them")
+ else()
+     if (EXIV2_ENABLE_XMP)
+-        find_package(EXPAT REQUIRED)
++        find_package(expat CONFIG REQUIRED)
++		set(EXPAT_FOUND 1)
```
Comment (JackBoosY) [inline ports/exiv2/fix-find_expat.patch:11]: Please convert the TABs to space then re-generate this patch.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**